src/Features/Core/Portable/TodoComments/ITodoCommentService.cs

```diff
@@ -46,12 +46,8 @@ public TodoComment(TodoCommentDescriptor descriptor, string message, int positio
                 priority: Descriptor.Priority,
                 message: Message,
                 documentId: document.Id,
-                originalLine: originalLineInfo.StartLinePosition.Line,
-                originalColumn: originalLineInfo.StartLinePosition.Character,
-                originalFilePath: document.FilePath,
-                mappedLine: mappedLineInfo.StartLinePosition.Line,
-                mappedColumn: mappedLineInfo.StartLinePosition.Character,
-                mappedFilePath: mappedLineInfo.GetMappedFilePathIfExist());
+                span: originalLineInfo,
+                mappedSpan: mappedLineInfo);
```

Comment:
don't need the GetMappedFilePathIfExist part here.  we can just pass the mappedLineInfo along.  We check GetMappedFilePathIfExist when converting to a DiagnosticDataLocation lower.